relay-metrics/src/aggregation.rs

```diff
@@ -1803,28 +1782,16 @@ impl Aggregator {
             let partitioned_buckets = self.partition_buckets(project_buckets, num_partitions);
             for (partition_key, buckets) in partitioned_buckets {
                 self.process_batches(buckets, |batch| {
-                    let fut = self
-                        .receiver
-                        .send(FlushBuckets {
-                            project_key,
-                            partition_key,
-                            buckets: batch,
-                        })
-                        .into_actor(self)
-                        .and_then(move |result, slf, _ctx| {
-                            if let Err(buckets) = result {
-                                relay_log::trace!(
-                                    "returned {} buckets from receiver, merging back",
-                                    buckets.len()
-                                );
-                                slf.merge_all(project_key, buckets).ok();
-                            }
-                            fut::ok(())
-                        })
-                        .drop_err();
-
-                    if let Some(context) = context.as_deref_mut() {
-                        fut.spawn(context);
+                    let batch_size = batch.len();
+                    match self.receiver.do_send(FlushBuckets {
+                        project_key,
+                        partition_key,
+                        buckets: batch,
+                    }) {
+                        Ok(_) => {}
+                        Err(err) => {
+                            relay_log::error!("Failed to flush the buckets: {err}. Dropping current bucket batch of size: {batch_size}");
```

Comment:
We should call `merge_all(...)` here such that the buckets are not dropped. `err` is a `SendError<FlushBuckets>` if I'm not mistaken, so we should be able to recover the buckets.

Reply:
Makes sense. 
I've added some changes in [1e10730](https://github.com/getsentry/relay/pull/1492/commits/1e10730568cb95f5ae7e0e4b6c6894c80484548d). PTAL!